MyRangesImplementation/MyRangesImplementation/cartezian_product_iterator.hpp

```diff
@@ -0,0 +1,112 @@
+#pragma once
+#include<type_traits>
+#include"range_traits.hpp"
+
+namespace ranges {
+	namespace internals {
+		namespace iterators {
+			template<typename... Iterators>
+			class cartezian_product_iterator {
```

Comment:
do you know Pimpl?
Do you want to use it here? is this a good case to use it here in your opinion?